src/stratis_cli/_actions/_top.py

```diff
@@ -172,27 +172,31 @@ def add_data_devices(namespace):
             .search(managed_objects)
         )
 
-        cache = [
-            MODev(info).Devnode()
+        blockdevs = frozenset(namespace.blockdevs)
+
+        cache = frozenset(
+            str(MODev(info).Devnode())
             for (_, info) in devs(props={"Tier": BlockDevTiers.Cache}).search(
                 managed_objects
             )
-        ]
-        already_cache = [dev for dev in namespace.blockdevs if dev in cache]
-        if already_cache != []:
-            raise StratisInUseError(already_cache, BlockDevTiers.Data)
+        )
+        already_cache = blockdevs.intersection(cache)
+        if list(already_cache) != []:
+            raise StratisInUseError(list(already_cache), BlockDevTiers.Data)
 
-        data = [
-            MODev(info).Devnode()
+        data = frozenset(
+            str(MODev(info).Devnode())
             for (_, info) in devs(props={"Tier": BlockDevTiers.Data}).search(
                 managed_objects
             )
-        ]
-        already_data = [dev for dev in namespace.blockdevs if dev in data]
+        )
+        already_data = blockdevs.intersection(data)
 
-        if already_data != []:
-            changed = [dev for dev in namespace.blockdevs if dev not in already_data]
-            raise StratisPartialChangeError("add-data", changed, already_data)
+        if list(already_data) != []:
+            changed = blockdevs.difference(already_data)
```

Comment:
```changed``` -> ```not_data``` ?

Reply:
I'm not sure that would be a clear variable name to me personally. I could see something like `new_data` for sure if that's acceptable?

Reply:
Resolved.